api/docker/caddy/Caddyfile

```diff
@@ -1,50 +1,76 @@
 {
-    # Debug
-    {$DEBUG}
+	# Debug
+	{$DEBUG}
 
-    http_port 3001
-    https_port 3443
-    auto_https disable_redirects
+	auto_https disable_redirects
 }
 
-{$SERVER_NAME}
-
-log
-
-# Matches requests for HTML documents, for static files,
-# except for known API paths and paths with extensions handled by API Platform
-@pwa expression `(
-        {header.Accept}.matches("\\btext/html\\b")
-        && !{path}.matches("(?i)(?:^/docs|^/graphql|^/bundles/|^/_profiler|^/_wdt|\\.(?:json|html$|csv$|ya?ml$|xml$))")
-    )
-    || {path} == "/favicon.ico"
-    || {path} == "/manifest.json"
-    || {path} == "/robots.txt"
-    || {path}.startsWith("/sitemap")`
-
-route {
-    root * /srv/api/public
-    mercure {
-        # Transport to use (default to Bolt)
-        transport_url {$MERCURE_TRANSPORT_URL:bolt:///data/mercure.db}
-        # Publisher JWT key
-        publisher_jwt {env.MERCURE_PUBLISHER_JWT_KEY} {env.MERCURE_PUBLISHER_JWT_ALG}
-        # Subscriber JWT key
-        subscriber_jwt {env.MERCURE_SUBSCRIBER_JWT_KEY} {env.MERCURE_SUBSCRIBER_JWT_ALG}
-        # Allow anonymous subscribers (double-check that it's what you want)
-        anonymous
-        # Enable the subscription API (double-check that it's what you want)
-        subscriptions
-        # Extra directives
-        {$MERCURE_EXTRA_DIRECTIVES}
-    }
-    vulcain
-    push
-
-    # Add links to the API docs and to the Mercure Hub if not set explicitly (e.g. the PWA)
-    header ?Link `</docs.jsonld>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation", </.well-known/mercure>; rel="mercure"`
-
-    php_fastcgi unix//var/run/php/php-fpm.sock
-    encode zstd gzip
-    file_server
+:3000 {
+	log {
+		level DEBUG
+	}
+
+	handle_path /api* {
+		# rewriting the uri used for php-fcgi did not work
+		# so we make a hop more to localhost:3001 with the rewritten url where the fcgi happens
+		# this may slow down the request
+
+		reverse_proxy localhost:3001 {
+			header_up X-Forwarded-Prefix "/api"
+		}
+	}
+
+	handle_path /print* {
+		reverse_proxy print:3003
```

Comment:
How about setting the X-Forwarded-Prefix for these other services as well? I could easily imagine the print or mail (or even frontend) service outputting an absolute link to another print or mail or frontend page (e.g. when serving an error response, the print service might link to some page with further info / logs, or the mail service might use links for navigation). For generating the correct hrefs, these services need to know the base URL under which they appear in the public.
I see that you tell the mail service about this via the MH_UI_WEB_PATH env variable. But it still wouldn't hurt to set the header here, in case we ever switch to a different dummy mail server.

Reply:
We could do it with print, but the print service currently does not use the header as far as i know.

The mail service gets the requests with /mail/.. (Thats the difference between handle_path and handle, handle does not rewrite the request).
mailhog does not use urls relative to the request url for its assets, that's why you have to tell mailhog under which path its reachable. If the request is not rewritten, the X-Forwared-Prefix Header does not make sense as far as i understand the header.